clippy_lints/src/trait_bounds.rs

```diff
@@ -238,31 +238,45 @@ fn check_trait_bound_duplication(cx: &LateContext<'_>, gen: &'_ Generics<'_>) {
         return;
     }
 
-    let mut map = FxHashMap::<_, Vec<_>>::default();
-    for predicate in gen.predicates {
+    let where_predicates = gen
+        .predicates
+        .iter()
+        .filter_map(|pred| {
```

Comment:
Please add a comment what this is actually filtering for. You can write example code in the comment and point to the parts this is looking for, like it is done for spans in error messages.

It may also be worth to define this `if_chain` in a closure above this `filter_map` and then use it here.

Reply:
Thanks for having a look @flip1995! Is this the only thing you want me to address?

Reply:
> It may also be worth to define this if_chain in a closure above this filter_map and then use it here.
Oof. I tried this the lifetimes were beyond me. Any tips or should I just go with the comment?

Reply:
I've added a comment on what is collected. Let me know if it sufficent. I made some small changes to make it more consistent with what currently exists as well.